sdk/java-sdk-protobuf/src/main/scala/kalix/javasdk/impl/action/ActionsImpl.scala

```diff
@@ -258,25 +265,32 @@ private[javasdk] final class ActionsImpl(_system: ActorSystem, services: Map[Str
   override def handleStreamedOut(in: ActionCommand): Source[ActionResponse, NotUsed] =
     services.get(in.serviceName) match {
       case Some(service) =>
-        try {
-          val context = createContext(in, service.messageCodec, None, service.serviceName)
-          val decodedPayload = service.messageCodec.decodeMessage(
-            in.payload.getOrElse(throw new IllegalArgumentException("No command payload")))
-          service.factory
-            .create(context)
-            .handleStreamedOut(in.name, MessageEnvelope.of(decodedPayload, context.metadata()), context)
-            .asScala
-            .mapAsync(1)(effect => effectToResponse(service, in, effect, service.messageCodec))
-            .recover { case NonFatal(ex) =>
-              // user stream failed with an "unexpected" error
-              handleUnexpectedException(service, in, ex)
+        // Note: invocation in future to guarantee create and invocation is running on sdk dispatcher with virtual thread support
+        Source
```

Comment:
question: without this we would run the `catch` bit on the default pool?

Reply:
Without this we would call `service.factory.create(context)` and then `handleStreamedOut` method on the returned instance, both invokes user logic which could be blocking, on the thread that called the surrounding `handleStreamedOut`, on default dispatcher (unless you tweak how the service is bound to an actual server listener, which we don't).